src/js/components/award/shared/overview/ProgressBar.jsx

```diff
@@ -0,0 +1,560 @@
+/**
+ * ProgressBar.jsx
+ * Created by Jonathan Hill 11/26/19
+ **/
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { scaleLinear } from 'd3-scale';
+import { pull, compact, uniqueId, isEmpty } from 'lodash';
+import RectanglePattern from 'components/sharedComponents/patterns/RectanglePattern';
+
+const propTypes = {
+    heightOfSVG: PropTypes.number,
+    heightOfProgressBar: PropTypes.number,
+    width: PropTypes.number,
+    domain: PropTypes.array,
+    milestones: PropTypes.array,
+    currentProgress: PropTypes.number,
+    awardType: PropTypes.string,
+    progressText: PropTypes.string,
+    textAdjustment: PropTypes.object,
+    badDomainData: PropTypes.bool,
+    descriptions: PropTypes.object
+};
+
+export default class ProgressBar extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            xScaleProgressBar: null,
+            xScaleWithinCircles: null,
+            currentProgress: 0,
+            thirdCircleData: 0,
+            badDomainData: false,
+            milestoneData: [],
+            showMilestones: false,
+            progressVerticalLineData: {},
+            progressBarPatternData: {},
+            progressTriangleData: '',
+            progressTextData: {}
+        };
+        this.textDiv = null;
+        // React will call this function when the DOM draws it ( React Callback Refs )
+        this.setTextDiv = (element) => {
+            this.textDiv = element;
+            this.positionText();
+        };
+    }
+
+    // recreate the progress bar on window resize
+    componentDidUpdate(prevProps) {
+        if (prevProps.width !== this.props.width) {
+            this.start();
+        }
+    }
+
+    start = () => {
+        this.validateDomainAndMilestones();
+        // DataFlow
+        // 1. start
+        // 2. validateDomainAndMilestones
+        // 2. createXScales
+        // 3. milestoneData
+        // 4. showMilestones
+        // 5. progressVerticalLineData
+        // 6. progressTriangleData
+        // 7. progressBarPatternData
+        // 8. progressTextData
+    }
+
+    validateDomainAndMilestones = () => {
+        const {
+            currentProgress,
+            domain
+        } = this.props;
+        // validation of domain
+        let badDomainData = this.props.badDomainData || false;
+        pull(domain, null, undefined, '');
+        if (domain.length !== 2) {
+            badDomainData = true;
+        }
+        if (domain[0] > domain[1]) badDomainData = true;
```

Comment:
might be helpful to use array destructiong here and give domain[0] a nice var name.

```javascript
const [startDate, endDate] = pull(domain, null, undefined, ''); 
```